draft-ietf-tls-external-psk-importer.md

```diff
@@ -106,19 +109,47 @@ structure as follows:
 [[TODO: An alternative design might combine label and hash into the same field so that future
 protocols which don't have a notion of HashAlgorithm don't need this field.]]
 
-ImportedIdentity.label MUST be bound to the protocol for which the key is imported. Thus,
-TLS 1.3 and QUICv1 {{!I-D.ietf-quic-transport}} MUST use "tls13" as the label. Similarly, TLS 1.2 and
+ImportedIdentity.label MUST be bound to the protocol into which the key is imported. Thus,
+TLS 1.3 and QUICv1 {{!I-D.ietf-quic-transport}} MUST use the label "tls13". Similarly, TLS 1.2 and
 all prior TLS versions should use "tls12" as ImportedIdentity.label, as well as SHA256 as ImportedIdentity.hash.
 Note that this means future versions of TLS will increase the number of PSKs derived from an external PSK.
 
 A unique and imported PSK (IPSK) with base key 'ipskx' bound to this identity is then computed as follows:
 
 ~~~
    epskx = HKDF-Extract(0, epsk)
-   ipskx = HKDF-Expand-Label(epskx, "derived psk",
-                             Hash(ImportedIdentity), Hash.length)
+   ipskx = HKDF-Expand-Label(epskx, "external psk importer",
+                             Hash(Context), Hash.length)
+~~~
+
+The `Context` in this case includes the `ImportedIdentity` plus any necessary additional security context information.
+
+~~~
+  struct {
+      ImportedIdentity imported_psk;
+      opaque client_id<0..2^16>;
+      PriorContext prior_contexts<0..2^16>;
+  } Context;
 ~~~
 
+The `client_id` is an optional field that is required to be unique for each actor that knows the the EPSK.
+This is to prevent Selfie-style reflections.
+This is only necessary in scenarios where more than two actors use the same key, including the case where a single agent will complete PSK handshakes as both the client and the server using the same key.
+
+~~~
+  struct {
+    opaque channel_binding<1..2^16>;
+    opaque secrets<0..2^16>;
+  } PriorContexts
+
+~~~
+
+`prior_contexts` is a list of prior security contexts, consisting of channel bindings and any associated keys.
+In the standard case this list will be empty, as the TLS connection is not wrapped in a prior security context.
+However if the OOB PSK was established through a protocol, or series of protocols, that provide a continuing security context, then including the channel binding for these contexts, as well as any keys established, binds the security contexts together.
+This makes it easier to reason formally about the exact properties are provided by the combined series of protocols.
```

Comment:
As above, I think descriptions would help. Perhaps we should add a section on OOB PSK deployments and how they might translate to different `Context` values?